### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by Tignass**
- created a array to centralize information for settings and variables
- added all the researches needed to launch rocket
- added a split on some building count (for example, split when there is 5 stone-furnaces)
- added some comments to the code
- added version 1.1.34-gog state
- completed the readme to fit new changes
- added a debug section to print some information in DebugView
- added a reset section which reset the timer if the current game is closed (will ease the upgrade and maintaining of the code)
- added the .CT file (for CheatEngine) to share the pointers found

### CODE REVIEW

**Comment by MrHug** [comment]
This looks amazing @Tignass!
I'll want to do a quick check to see if everything is working on my end as well, hope to have some time for that soon!

**Comment by Tignass** [comment]
Great, I'm happy you like it :)

I attached the .CT file (for cheat engine) if you want to the check easily.

File: Factorio.asl [hunk 1]
```diff
@@ -117,25 +362,64 @@ start {
...
-	if (settings["research"]) {
-		foreach(var science in vars.researches.Keys){
-			if (!settings[science]) continue;
-			foreach(KeyValuePair<string, ulong> entry in vars.researches[science]) {
-				if (settings[entry.Key] && !vars.splitsDone.Contains(entry.Key)) { // We want to split on it and it hasn't happened yet.
-					if (memory.ReadValue<byte>((System.IntPtr)(current.researchArea + entry.Value)) == 1) {
-						vars.splitsDone.Add(entry.Key);
-						return true;
+	
+	// will split when searches are completed (for the ones selected in the Auto-splitter settings)
```
Comment (MrHug) [inline Factorio.asl:373]:
```suggestion
	// will split when researches are completed (for the ones selected in the Auto-splitter settings)
```

File: Factorio.asl [hunk 2]
```diff
@@ -146,3 +430,8 @@ isLoading {
 shutdown {
 	vars.splitsDone = new HashSet<string>();
 }
+reset {
+	if(memory.ReadValue<byte>((System.IntPtr)(current.gamePointer)) == 0){
```
Comment (MrHug) [inline Factorio.asl:434]: I was wondering about whether or not we should include this. Perhaps we can make the default "reset" value in the livesplitter false? (If that is even possible?)

At least when I run the coop Any% we sometimes have a disconnect, and if that resets a timer that would be very unfortunate... ;(

**Review by MrHug — COMMENTED** [review]
Alright, I've taken a look, still amazed by all the lovely stuff you found here! 😍 

Couple of things to either look at now, or to include in the future:

- In the .CT file, you can make all the building ones relative to the base pointer. What I suggest is make an entry: "BuildingArea" and then you can set up other adresses relative to that. That way we need to only update 1 pointer for a new version (coincidentally for the latest steam version, the pointer is: `"factorio.exe"+01C1F940`) instead of for every entry.
image
image
- There is a typo in your split names (`Buner` instead of `Burner`) as well as some inconsistency in the styling (hyphenated splits names vs Capital Letters vs Only one capital). Not a big deal, but an easy fix and a probably a nice fix too ;)
- Long long term: In your estimation, to what extent do you think it's feasible to make the number of buildings you split on configurable through the livesplit interface? From what I recall support for that is not existent?
- Should we make an `Authors.md` file, or have a list of contributors in the README now? :)

I'm happy to already merge as is, but figured I'd share the feedback with you first if you want to take a look :) Let me know what you want/think you can fix! :D

Thanks again!

**Comment by aureianimus** [comment]
I went ahead and fixed some of the typo's, the reset stuff seems fine for now, I think. We can make the CT file more dynamic later and this is way more awesome than it used to be.

**Comment by Tignass** [comment]
Hey,

Sorry, I put this project on the side, and probably will for (at least) this summer...

Thank you for checking my code, correcting the mistakes and merging it :)

If you want, I started a discussion on  [speedrun.com](https://www.speedrun.com/factorio/thread/erf6j) to have some feedback from speed-runners.

Have a nice summer.

**Comment by MrHug** [comment]
No worries, all the work you'd already put in was amazing :)
 Enjoy your summer too!

**Comment by Tignass** [comment]
I received a message on reddit from an user wanting to use our auto-splitter, he had no clue on how to set it up.

I think a tutorial section in the readme file would be great so more speerunner can use it (and help us improve it).

Here is my answer to him, i'm not sure it's clear enough...:

```
Hi Rain,

Sorry I didn't see your message earlier.

You can find the autosplitter on https://github.com/aureianimus/LiveSplit.Factorio

There is no tutorial yet but I'll try give you indications on how to use it (it might not be 100% correct as I'm in holidays on my linux laptop - can't install livesplit)

-   First you need to install: https://livesplit.org/downloads/

-   Download : https://github.com/aureianimus/LiveSplit.Factorio/archive/refs/heads/master.zip
-   Unzip
-   In livesplit open the .iss (layout file)
-   Add the Scriptable Auto Splitter component to the Layout. Right-click on LiveSplit and choose "Edit Layout..." to open the Layout Editor,
- -  then click on the Plus-sign and choose "Scriptable Auto Splitter" from the section "Control". (if not already present in the layout file)

-   Choose the path of the script (.asl) 
- - To get to the settings of the component you can either double click it in the Layout Editor 
- - or go into to the Scriptable Auto Splitter Tab of the Layout Settings. Once you've set the Path, the script should automatically load and hopefully work.

In the settings, you can select the options for the auto splitting (on which research or on building count).

If you want to change the building count autosplit, you can edit the .asl between line 222 and line 324. 
For now there is only 5 auto-splits for building count (Burner City, First power, Green chips, 36 labs and Blue Science) 
based on "Nefrums' Factorio any% speedrun guide"
https://docs.google.com/presentation/d/19yuwxdcBKODTlfo03ArbNa0_4Qtgz--IOaA-dGU-RZY/edit 
If you'd like to implement more, it will be really appreciated.

If you find any bug or if you see any improvement to be made, pls report it on https://github.com/aureianimus/LiveSplit.Factorio/issues/new

If you want to help with this project, a tutorial on how to use it (in text with screenshots or video) 
would really be appreciated (if my explications where clear enough for you to use it).

Do not hesitate to contact me again if you have any trouble or questions.

Cheers
```

What do you think ?

**Pull request merged**